r/R/table.R

```diff
@@ -254,6 +257,49 @@ names.Table <- function(x) x$ColumnNames()
 #' @export
 `[[.Table` <- `[[.RecordBatch`
 
+#' @export
+`[[<-.Table` <- function(x, i, value) {
+  if (is.null(value)) {
+    if (is.character(i)) {
+      i <- match(i, names(x)) - 1L
+    }
```

Comment:
I think we need to do the `- 1L` even if i is not character, otherwise we'll be off by one with an integer `i`.